gremlin-go/driver/resultSet.go

```diff
@@ -120,6 +121,19 @@ func (channelResultSet *channelResultSet) Close() {
 	}
 }
 
+// Close and remove from the channelResultSet from the container without locking container. Meant for use when calling
```

Comment:
Is `Close` still used? Overall this LGTM and I think solves this well - just wondering if we can remove the other Close() now.

Reply:
If it is used, perhaps we should have
`LockedClose()`
and
`unlockedClose()`

and both should call 
```
func (channelResultSet *channelResultSet) close(locked bool) {
	if !channelResultSet.closed {
		channelResultSet.channelMutex.Lock()
		channelResultSet.closed = true
		if lock {
     			channelResultSet.container.delete(channelResultSet.requestID)
		} else {
			delete(channelResultSet.container.internalMap, channelResultSet.requestID)
		}
		close(channelResultSet.channel)
		channelResultSet.channelMutex.Unlock()
		channelResultSet.sendSignal()
	}
}```

Reply:
Close is still used in the happy path where a ResultSet is fully read, as well as being an exposed point for the user to mark it as completed. This makes it hard for us to modify the API around that.